src/core/service.c

```diff
@@ -2588,7 +2588,9 @@ static int service_deserialize_item(Unit *u, const char *key, const char *value,
                         log_unit_debug(u, "Failed to parse main-pid value: %s", value);
                 else {
                         service_set_main_pid(s, pid);
-                        unit_watch_pid(UNIT(s), pid);
+                        r = unit_watch_pid(UNIT(s), pid);
+                        if (r < 0)
+                                log_unit_debug_errno(u, r, "Failed to watch main PID, ignoring: %m");
```

Comment:
hmm, _debug.  I see parse failures are all logged as _debug, but there's a couple of _error's for things that look like a clear ENOMEM.  This PR seems to be prep related to #7860.  _after_ #7860, unit_watch_pid() should stop failing if too many different units watch the same pid, and should only fail for ENOMEM, right?  But I didn't find a change of this line in #7860.  Should #7860 upgrade this message from _debug to `log_unit_error_errno()`?

Reply:
Good point, but I think this can be fixed later.